src/adapters/adyoulike.js

```diff
@@ -0,0 +1,180 @@
+import Adapter from 'src/adapters/adapter';
+import bidfactory from 'src/bidfactory';
+import bidmanager from 'src/bidmanager';
+import * as utils from 'src/utils';
+import { format } from 'src/url';
+import { ajax } from 'src/ajax';
+import { STATUS } from 'src/constants';
+
+var AdyoulikeAdapter = function AdyoulikeAdapter() {
+  const _VERSION = '0.1';
+
+  const baseAdapter = Adapter.createNew('adyoulike');
+
+  baseAdapter.callBids = function (bidRequest) {
+    const bidRequests = {};
+    const bids = bidRequest.bids || [];
+
+    const validBids = bids.filter(valid);
+    validBids.forEach(bid => { bidRequests[bid.params.placement] = bid; });
+
+    const placements = validBids.map(bid => bid.params.placement);
+    if (!utils.isEmpty(placements)) {
+      const body = createBody(placements);
+      const endpoint = createEndpoint();
+      ajax(endpoint,
+        (response) => {
+          handleResponse(bidRequests, response);
+        }, body, {
+          contentType: 'text/json',
+          withCredentials: true
+        });
+    }
+  };
+
+  /* Create endpoint url */
+  function createEndpoint() {
+    return format({
+      protocol: (document.location.protocol === 'https:') ? 'https' : 'http',
+      host: 'hb-api.omnitagjs.com',
+      pathname: '/hb-api/prebid',
+      search: createEndpointQS()
+    });
+  }
+
+  /* Create endpoint query string */
+  function createEndpointQS() {
+    const qs = {};
+
+    const ref = getReferrerUrl();
+    if (ref) {
+      qs.RefererUrl = encodeURIComponent(ref);
+    }
+
+    const can = getCanonicalUrl();
+    if (can) {
+      qs.CanonicalUrl = encodeURIComponent(can);
+    }
+
+    return qs;
+  }
+
+  /* Create request body */
+  function createBody(placements) {
+    const body = {
+      Version: _VERSION,
+      Placements: placements,
+    };
+
+    if (performance && performance.navigation) {
+      body.PageRefreshed = performance.navigation.type === performance.navigation.TYPE_RELOAD;
+    }
+
+    return JSON.stringify(body);
+  }
+
+  /* Response handler */
+  function handleResponse(bidRequests, response) {
+    let responses = [];
+    try {
+      responses = JSON.parse(response);
+    } catch (error) { utils.logError(error); }
+
+    const bidResponses = {};
+    responses.forEach(response => {
+      bidResponses[response.Placement] = response;
+    });
+
+    Object.keys(bidRequests).forEach(placement => {
+      addResponse(placement, bidRequests[placement], bidResponses[placement]);
+    });
+  }
+
+  /* Check that a bid has required parameters */
+  function valid(bid) {
+    const sizes = getSize(bid.sizes);
+    if (!bid.params.placement || !sizes.width || !sizes.height) {
+      return false;
+    }
+    return true;
+  }
+
+  /* Get current page referrer url */
+  function getReferrerUrl() {
+    try {
+      return window.top.document.referrer;
+    } catch (e) { }
+    return document.referrer;
```

Comment:
Generally, it's frowned upon to use try/catch for control flow: https://softwareengineering.stackexchange.com/a/189225

You can write this linearly by testing for undefined (the same goes for `getCanonicalUrl`).